### PR TITLE
dstore/pmix: flock portability

### PR DESCRIPTION
**Pull request by karasevb**
Fixed the problem introduced in #136.

@ggouaillardet, @artpol84 please review.

### CODE REVIEW

File: config/pmix.m4 [hunk 1]
```diff
@@ -659,6 +659,14 @@ AC_DEFUN([PMIX_SETUP_CORE],[
     AC_SUBST(pmixlibdir)
     AC_SUBST(pmixincludedir)
 
+    ##################################
+    # Flock
+    ##################################
+    pmix_show_subtitle "Configuring flock"
+
+    AC_CHECK_FUNC([flock], [AC_DEFINE([PMIX_ENABLE_FLOCK], [ ],
```
Comment (ggouaillardet) [inline config/pmix.m4]: or you can add ```flock``` to the arguments of ```AC_CHECK_FUNCS``` and test with
```c
#ifdef HAVE_FLOCK
...
#endif
```

**Comment by artpol84** [comment]
Thank you.
Can you verify that it works ok on Solaris as is?
And then we can address the comment.

**Comment by ggouaillardet** [comment]
Will do tomorrow

**Comment by artpol84** [comment]
Here we obviously will not sync procs from several nodes using it. But I think we still might need NFS support to sync procs on the same node.

**Comment by ggouaillardet** [comment]
i was able to build pmix on Solaris, i needed #151 though (not critical, since only test and example are affected)

**Comment by artpol84** [comment]
@ggouaillardet thank you.

To close the question on NFS locking here is the excerpt from the `flock` man page:
```
       In Linux kernels up to 2.6.11, flock() does not lock files over NFS (i.e., the scope of locks was limited to the local system).  Instead, one could use fcntl(2) byte-range locking, which does work over NFS, given a sufficiently recent version of Linux and a server which supports locking.  Since Linux 2.6.12, NFS clients support flock() locks by emulating them as byte-range locks on the entire file.  This means that fcntl(2) and flock() locks do interact with one another over NFS.  Since Linux 2.6.37, the kernel supports a compatibility mode that allows flock() locks (and also fcntl(2) byte region locks) to be treated as local; see the discussion of the local_lock option in nfs(5).
```
so `flock` is fine for NFS in recent kernels. However I agree that since we can achieve the same effect with both ways. But `fcntl` is known to be more portable and have wider coverage in FS so let's go ahead with it.

**Comment by artpol84** [comment]
All we need from NFS (if the dstor session dir will be located there) is `local_lock` to allow the locking mechanism that we are using.

**Comment by karasevb** [comment]
@artpol84 please review

**Comment by artpol84** [comment]
Looks OK.
@ggouaillardet could you double-check before we merge?

**Comment by ggouaillardet** [comment]
that looks good to me too !

**Comment by artpol84** [comment]
@ggouaillardet thank you!

**Comment by artpol84** [comment]
@karasevb please, PR to the v2.x

**Pull request merged**